docs/modules/ROOT/pages/api/erc20.adoc

```diff
@@ -0,0 +1,265 @@
+
+[.contract-item]
+[[IERC20-balance_of]]
+==== `[.contract-item-name]#++balance_of++#++(account: ContractAddress) → u256++` [.item-kind]#external#
+
+Returns the amount of tokens owned by `account`.
+
+[.contract-item]
+[[IERC20-allowance]]
+==== `[.contract-item-name]#++allowance++#++(owner: ContractAddress, spender: ContractAddress) → u256++` [.item-kind]#external#
+
+Returns the remaining number of tokens that `spender` will be allowed to spend on behalf of `owner` through <<transfer_from,transfer_from>>. This is zero by default.
+
+This value changes when <<approve,approve>> or <<transfer_from,transfer_from>> are called.
+
+[.contract-item]
+[[IERC20-transfer]]
+==== `[.contract-item-name]#++transfer++#++(recipient: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+Moves `amount` tokens from the caller's token balance to `to`.
+Returns `true` on success.
+
+Emits a <<Transfer,Transfer>> event.
+
+This function panics if:
+
+- The caller is the zero address.
+- `recipient` is the zero address.
+
+[.contract-item]
+[[IERC20-transfer_from]]
+==== `[.contract-item-name]#++transfer_from++#++(sender: ContractAddress, recipient: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+Moves `amount` tokens from `sender` to `recipient` using the allowance mechanism.
+`amount` is then deducted from the caller's allowance.
+Returns `true` on success.
+
+Emits a <<Transfer,Transfer>> event.
+
+This function panics if:
+
+- `sender` is the zero address.
+- `recipient` is the zero address.
+
+[.contract-item]
+[[IERC20-approve]]
+==== `[.contract-item-name]#++approve++#++(spender: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+Sets `amount` as the allowance of `spender` over the caller's tokens.
+Returns `true` on success.
+
+Emits an <<Approval,Approval>> event.
+
+This function panics if:
+
+- The caller is the zero address.
+- `spender` is the zero address.
+
+[.contract-item]
+[[IERC20-Transfer]]
+==== `[.contract-item-name]#++Transfer++#++(from: ContractAddress, to: ContractAddress, value: u256)++` [.item-kind]#event#
+
+Emitted when `value` tokens are moved from one address (`from`) to another (`to`).
+
+Note that `value` may be zero.
+
+[.contract-item]
+[[IERC20-Approval]]
+==== `[.contract-item-name]#++Approval++#++(owner: ContractAddress, spender: ContractAddress, value: u256)++` [.item-kind]#event#
+
+Emitted when the allowance of a `spender` for an `owner` is set by a call to <<approve,approve>>.
+`value` is the new allowance.
+
+[.contract]
+[[ERC20]]
+=== `++ERC20++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/token/erc20/erc20.cairo[{github-icon},role=heading-link]
+
+[.hljs-theme-light.nopadding]
+```javascript
+use openzeppelin::token::erc20::ERC20;
+```
+
+Implementation of the <<IERC20,IERC20>> interface.
+
+[.contract-index]
+.Functions
+--
+* xref:#constructor[`++constructor(name: felt252, symbol: felt252, initial_supply: u256, recipient: ContractAddress)++`]
```

Comment:
As I said, I think it may be a good idea to add constructors, but I'm having second thoughts because they are not really part of the API (are not external methods, or internal), and they are not included in the Solidity docs version. IMO is better to design a way of presenting this that is not in the functions section, I'm leaning towards leaving this for a future iteration as well, wdty?

Reply:
>  I'm having second thoughts because they are not really part of the API (are not external methods, or internal)

We could set the `[.item-kind]` to `constructor` on the right side. If we like this, I think we should make it a point to always include it as the first function on the `Functions` list

> they are not included in the Solidity docs version

They are included though, no? https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#ERC20-constructor-string-string-

> IMO is better to design a way of presenting this that is not in the functions section, I'm leaning towards leaving this for a future iteration as well,  wdty?

Hmm I'm actually leaning the other way. If we're insistent that the `constructor` doesn't fit in `Functions`, I think it's better to include it temporarily so users can still find information on it as opposed to completely omitting it. Wdyt?

Reply:
I totally agree. Old message sorry, I got to the same point after a couple of iterations.

Reply:
Haha no worries, glad we're in agreement